portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,13 +40,33 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String maxCommentsString = request.getParameter("number-comment-choice");
+
+    int maxComments;
+    try {
+      maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxCommentsString);
+      maxComments = 0;
+    }
+
     List<String> messages =  new ArrayList<>();
     Query query = new Query("Task").addSort("text-comment", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
     for (Entity entity : results.asIterable()) {
-      String a = (String) entity.getProperty("text-comment");
-      messages.add(a);
+      if (maxComments > 0) {
```

Comment:
This works but it's error-prone. You're using maxComments as a counter and future editors wouldn't expect. They could use in a for loop later without carefully reading the code and not realize that it's 0 and not the actual max number of comments as its name suggests.

Also, you loop over all results even if maxComments is set to 1. So if there are 1000 comments, your servlet does a bunch of useless work.

It would be better if you create a new counter or use the size of the array as a counter
`if (messages.size() >= maxComments) { break; }`

The break exits the for loop

Reply:
Done!